torch/_dynamo/variables/builder.py

```diff
@@ -885,18 +891,25 @@ def wrap_unspecialized_primitive(self, value):
 
                 name = self.source.name()
                 if name not in self.tx.output.frame_state:
-                    curr_size = value
+                    # Note - this esentially means that if this name gets reused as a tensor,
+                    # it will start fully dynamic. That should always be a safe option, and not awfully inefficient.
+                    # Alternatively, if we want to improve pef here, we can add a third state of unset, but I am not
+                    # sure that is necessary for now.
+                    frame_state_entry = FrameStateSizeEntry(scalar=value, size=None)
                 else:
-                    curr_size = self.tx.output.frame_state[name]
-                    if curr_size != value:
-                        curr_size = None
-                self.tx.output.frame_state[name] = curr_size
+                    frame_state_entry = self.tx.output.frame_state[name]
+                    if frame_state_entry.scalar != value:
+                        frame_state_entry.scalar = None
+                self.tx.output.frame_state[name] = frame_state_entry
```

Comment:
If the struct was already set you technically didn't need this

Reply:
Yes, but I like the readability this offers.